src/Controller/ApiV1Controller.php

```diff
@@ -0,0 +1,211 @@
+<?php
+/**
+ * Tatoeba Project, free collaborative creation of multilingual corpuses project
+ * Copyright (C) 2009-2010  HO Ngoc Phuong Trang <tranglich@gmail.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * PHP version 5
+ *
+ * @category PHP
+ * @package  Tatoeba
+ * @author   HO Ngoc Phuong Trang <tranglich@gmail.com>
+ * @license  Affero General Public License
+ * @link     https://tatoeba.org
+ */
+namespace App\Controller;
+
+use App\Controller\AppController;
+use App\Form\SentencesSearchForm;
+use App\Model\CurrentUser;
+use App\Model\Table\SentencesTable;
+use App\Lib\LanguagesLib;
+use App\Lib\SphinxClient;
+use App\Lib\Licenses;
+use Cake\Core\Configure;
+use Cake\Database\Expression\QueryExpression;
+use Cake\Event\Event;
+use Cake\Routing\Router;
+use Cake\Utility\Hash;
+use Cake\View\ViewBuilder;
+use Exception;
+
+/**
+ * Controller for Api.
+ *
+ * @category Sentences
+ * @package  Controllers
+ * @author   HO Ngoc Phuong Trang <tranglich@gmail.com>
+ * @license  Affero General Public License
+ * @link     https://tatoeba.org
+ */
+class ApiV1Controller extends AppController
+{
+    public $name = 'Sentences';
+    public $components = array (
+        'LanguageDetection',
+        'CommonSentence',
+        'Permissions',
+    );
+    public $helpers = array(
+        'Sentences',
+        'Menu',
+        'Lists',
+        'SentenceButtons',
+        'Html',
+        'Logs',
+        'Pagination',
+        'Comments',
+        'Languages',
+        'CommonModules'
+    );
+
+    public $uses = array(
+        'Audio',
+        'Sentence',
+        'SentencesSentencesLists',
+        'SentencesList',
+        'User',
+        'UsersLanguages',
+        'Tag',
+        'UsersSentences',
+        'Vocabulary'
+    );
+
+    public function initialize() {
+        parent::initialize();
+
+        $params = $this->request->params;
+        $noCsrfActions = ['edit_sentence', 'change_language'];
+        if (in_array($params['action'], $noCsrfActions)) {
+            $this->components()->unload('Csrf');
+        }
+    }
+
+    /**
+     * Show Json of specified sentence id.
+     * 
+     * @param mixed $id Id of the sentence.
+     * 
+     * @return void
+     */
+    public function sentence($id = null) {
```

Comment:
I suggest renaming this method from `sentence` to `sentences`. Using plural in this situation is a well-known API naming convention.

Reply:
Meh, on this one I'm currently not so hot about changing it. 

https://nordicapis.com/10-best-practices-for-naming-api-endpoints/ says:
![image](https://user-images.githubusercontent.com/7943077/117717638-5a8bd600-b1db-11eb-9edc-abbaf4a7b76c.png)

I think the "Admittedly this is a matter of preference" suggests it depends. I certainly think "sentence/5" (implying sentence number 5) rather than "sentences/5" (implying 5 sentences?) appeals more to me.

Reply:
Yeah it’s probably a matter of perspective. The way I see "/sentences/5" is "number 5 inside the directory of sentences". I don’t think we need to argue on such detail at that point; just leave it as you want.